packages/hash/api/src/graph/index.ts

```diff
@@ -13,6 +13,11 @@ import {
 import { ensureSystemTypesExist } from "./system-types";
 import { ensureSystemEntitiesExists } from "./system-entities";
 
+export type GraphContext = {
+  graphApi: GraphApi;
+  /** @todo: add logger? */
+};
```

Comment:
I thought this might a good point in time to introduce some kind of context object that passes things like the `graphApi` (and perhaps the `logger` down the line) to the functions, similar to how each GQL resolver receives a context object. This would mean that adding/removing something wouldn't require redefining the types of all the functions, and it might help us be more consistent. Keen to get thoughts on this, and also don't feel particularly strongly about it so am happy to just go back to passing `graphApi` as a top-level function param.

Reply:
We've partially discussed this offline, the new `Impure` and `PureGraphFunction` types explicitly mark the functions now. I don't think we need to worry about moving them out to another folder/module yet as it's very clear whether or not we're dealing with side effects